src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -456,13 +449,6 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
                 {
                     var diagnostics = diagnosticsMap[analyzerWithState.Analyzer];
                     builder.AddRange(diagnostics.Where(ShouldInclude));
-
-                    // Save the computed diagnostics if caching is enabled and diagnostics were computed for the entire document.
-                    if (_cacheFullDocumentDiagnostics && !span.HasValue)
-                    {
-                        var data = new DocumentAnalysisData(version, _text.Lines.Count, diagnostics);
-                        analyzerWithState.State.Save(executor.AnalysisScope.Kind, data);
-                    }
```

Comment:
here's why i think this is ok to remove.

1. this code doesn't run in normal VS scenarios (where everything is working fine).
2. it *only* runs in LSP, and LSP is the only case where things are buggy.  So we should just align both on the behavior that is non-buggy.

Reply:
If we remove this -  anything else calling `State.Save`?  It looks like some event on document open and [AnalyzeDocumentForKindAsync](https://sourceroslyn.io/Microsoft.CodeAnalysis.LanguageServer.Protocol/R/cb0a140fe44c3bde.html)

Its a bit hard to track down but is that only from the sln crawler?

Reply:
i'll see if i can clean this up more.  but i wouldn't mind the PR being minimal.

Reply:
What is the root cause of stale diagnostics here?

Reply:
no clue.  but we've confirmed on the LSP side that we're passing in the right document snapshot :)   so something *within* the diagnostics layer is caching and causing problems.  

As this is not a problem outside of LSP (where we are not doing the caching) and only a problem within LSP (where we are currently not doing it), the easiest thing was to just stop doing it :)

if you want to investigate the caching problem, i'm def ok with that.  but i'd like us to get into a good state first :)